### ISSUE TITLE
How to calculate confusion matrix and average precision for each object class using custom dataset?

### ISSUE THREAD

**Issue by UmmarHassan96** [facebookresearch/detr#477]
## ❓ How to do something using DETR

Describe what you want to do, including:
1. what inputs you will provide, if any:
2. what outputs you are expecting:

NOTE:

1. Only general answers are provided.
   If you want to ask about "why X did not work", please use the
   [Unexpected behaviors](https://github.com/facebookresearch/detr/issues/new/choose) issue template.

2. About how to implement new models / new dataloader / new training logic, etc., check documentation first.

3. We do not answer general machine learning / computer vision questions that are not specific to DETR, such as how a model works, how to improve your training/make it converge, or what algorithm/methods can be used to achieve X.

**Comment by L-xiaolu** [comment]
Did you solve this problem please?

**Comment by L-xiaolu** [comment]
> Did you solve this problem?

建议使用mmdetection的混淆矩阵工具

**Comment by UmmarHassan96** [comment]
yes i solved it first train model using resnet 50 not resnet 100 then convert the model using d2 folder  using converter.py and then use train_net.py with --eval-only to calculate mAP per object

**Comment by nicollecnunes** [comment]
did you solve this problem?
